### PR TITLE
Copy custom measures to analysis

### PR DESCRIPTION
**Pull request by simonbrandhof**
Custom measures that are defined by end-users are copied to analysis report by compute engine.

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/CustomMeasuresCopyStep.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      public void visitAny(Component component) {
+        copy(component);
+      }
+    }.visit(treeRootHolder.getRoot());
+  }
+
+  private void copy(Component component) {
+    for (CustomMeasureDto dto : loadCustomMeasures(component)) {
+      Metric metric = metricRepository.getById(dto.getMetricId());
+      // TODO verify that deletion of custom metrics correctly delete related measures,
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/CustomMeasuresCopyStep.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/CustomMeasuresCopyStep.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    DbSession session = dbClient.openSession(false);
+    try {
+      return dbClient.customMeasureDao().selectByComponentUuid(session, component.getUuid());
+    } finally {
+      MyBatis.closeQuietly(session);
+    }
+  }
+
+  @VisibleForTesting
+  static Measure dtoToMeasure(CustomMeasureDto dto, Metric metric) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/CustomMeasuresCopyStep.java]: MAJOR The Cyclomatic Complexity of this method "dtoToMeasure" is 13 which is greater than 10 authorized. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
only taste comments, LGTM

**Comment by simonbrandhof** [comment]
All feedback applied. Thanks

**Pull request closed without merging**